packages/flutter/test/material/tabs_test.dart

```diff
@@ -2350,4 +2350,94 @@ void main() {
     expect(find.text('Tab1'), findsOneWidget);
     expect(find.text('Tab2'), findsOneWidget);
   });
+
+    testWidgets('DefaultTabController with zero tabs',
+          (WidgetTester tester) async {
+        // Regression test for https://github.com/flutter/flutter/issues/32428
+        int noOfTabs = 0;
+        List<Widget> tabList;
+        List<Widget> tabViewList;
+
+        await tester.pumpWidget(
+          MaterialApp(
+            home: StatefulBuilder(
+              builder: (BuildContext context, StateSetter setState) {
+                return DefaultTabController(
+                    length: noOfTabs,
+                    child: Scaffold(
+                      appBar: AppBar(
+                        title: const Text('Default TabBar Preview'),
+                        bottom: (noOfTabs > 0)
+                            ? TabBar(isScrollable: true, tabs: tabList)
+                            : null,
+                      ),
+                      body: (noOfTabs > 0)
+                          ? TabBarView(children: tabViewList)
+                          : const Center(
+                        child: Text('No tabs'),
+                      ),
+                      bottomNavigationBar: BottomAppBar(
+                        child: Row(
+                          mainAxisSize: MainAxisSize.max,
+                          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                          children: <Widget>[
+                            IconButton(
+                              key: const Key('Add tab'),
+                              icon: const Icon(Icons.add),
+                              onPressed: () {
```

Comment:
This logic can be further simpified if you're simply adding a new item to your list:
```
                          onPressed: () {
                            setState(() {
                              tabList.add(Tab(text: 'Tab ${tabList.length + 1}'));
                              tabViewList.add(Center(child: Text('Tab ${tabViewList.length + 1}')));
                            });
                          },
```

Reply:
Also, it's better practice to have your state change logic happen within `setState`